### ISSUE TITLE
Unable to deform certain chunks

### ISSUE THREAD

**Issue by ThisUserIsNobody** [Eldemarkki/Marching-Cubes-Terrain#41]
After I switched the terrain to generate using 3D noise, I encountered deformation issues. Some chunks are not deformable. When I try to deform the chunk terrain becomes darker but nothing gets changed. They entire chuck seems like flickers or something. And other chunks works as they should. It's really strange..

![issue](https://user-images.githubusercontent.com/67506162/87584796-b59d8b00-c6e6-11ea-8efd-8fd90d3250e5.jpg)

**Comment by Eldemarkki** [comment]
That is very strange indeed... did you change anything else anywhere in the code (or the scene) other than the `CalculateVoxelData()` function? Because changing that function should not break anything.

**Comment by Eldemarkki** [comment]
I've managed to reproduce a similar type of bug, and I believe that what you are seeing is caused by it. It happens when you repeatedly go back and forward between chunks, causing the generation of new chunks/removal of old chunks, but the new chunks don't get registered/old chunk's don't get removed visually.

**Comment by Eldemarkki** [comment]
I created a hotfix to the master branch that should fix this. Could you please try and see if you can still reproduce the bug after pulling the changes?

**Comment by ThisUserIsNobody** [comment]
Will take a look!

**Comment by ThisUserIsNobody** [comment]
After I updated the project I started to receive this error message whenever I would press play on both demo scenes (heightmap and procedural).

`Unexpected exception Burst.Compiler.IL.CompilerException: Error while verifying module: Call parameter type does not match function signature!
  %26 = load i8, i8* %var.2
 i32  call void @"$___check_bounds"(i8 %26, i32 256)

  at Burst.Compiler.IL.NativeCompiler.Compile () [0x001d9] in <0a91b9ecc469494eb1ac29fa079c9115>:0 
  at Burst.Compiler.IL.Jit.JitCompiler.CompileMethodInternal (Burst.Compiler.IL.Jit.JitResult result, System.Collections.Generic.List`1[T] methodsToCompile, Burst.Compiler.IL.Jit.JitOptions jitOptions) [0x00224] in <0a91b9ecc469494eb1ac29fa079c9115>:0 
  at Burst.Compiler.IL.Jit.JitCompiler.CompileMethods (Mono.Cecil.MethodReference[] methodReferences, Burst.Compiler.IL.Jit.JitOptions jitOptions) [0x00254] in <0a91b9ecc469494eb1ac29fa079c9115>:0 
  at Burst.Compiler.IL.Jit.JitCompiler.CompileMethod (Mono.Cecil.MethodReference methodReference, Burst.Compiler.IL.Jit.JitOptions jitOptions) [0x0001c] in <0a91b9ecc469494eb1ac29fa079c9115>:0 
  at Burst.Compiler.IL.Jit.JitCompilerService+CompilerThreadContext.Compile (Burst.Compiler.IL.Jit.JitCompilerService+CompileJob job) [0x00396] in <0a91b9ecc469494eb1ac29fa079c9115>:0 

While compiling job: System.Void Unity.Jobs.IJobParallelForExtensions/ParallelForJobStruct`1<Eldemarkki.VoxelTerrain.Meshing.MarchingCubes.MarchingCubesJob>::Execute(T&,System.IntPtr,System.IntPtr,Unity.Jobs.LowLevel.Unsafe.JobRanges&,System.Int32)
at <empty>:line 0
`

**Comment by Eldemarkki** [comment]
Uh oh. Could you post the CalculateVoxelData and OctaveNoise functions here? Also, did you do any other changes to the code (you can list the files with changes with `git diff --name-only`)

**Comment by ThisUserIsNobody** [comment]
So I managed to fix this issue, by upgrading the burst package and re-importing the libraries. I'm not receiving any errors anymore. The burst package needed to be in version 1.3.3. It was required by the collection package which was imported by the jobs system.

I used the same noise settings and have generated a 3D terrain. Did some playtesting, but wasn't able to reproduce the previous deformation bug, so I believe it's fixed!
